R/getComps.R

```diff
@@ -1,127 +1,156 @@
-    c(usualSuspects, 'lengthcm', "Age")))
+    c(usualSuspects, "lengthcm", "Age")
+  ))
   usualSuspects <- switch(Comps,
     LEN = c(usualSuspects, "lengthcm"),
     AGE = c(usualSuspects, "Age"),
-    c(usualSuspects, "lengthcm", "Age"))
-
-  if (verbose) {
-    cat("\nAggregating, stratification is by", paste(c(strat, usualSuspects), collapse=", "), "\n\n")
-    utils::flush.console()
-  }
+    c(usualSuspects, "lengthcm", "Age")
+  )
 
-  ageComps <- getcomps_long(data = Pdata,
-    towstrat = TowStrat, type = utils::tail(usualSuspects, 1), ...)
+  ageComps <- getcomps_long(
+    data = Pdata,
+    towstrat = TowStrat,
+    type = utils::tail(usualSuspects, 1),
+    ...
+  )
   invisible(ageComps)
+}
 
-} # End function getComps
-
-#' Create a Long Database to Prepare Compositions
-#' @param data A data frame with columns defined by the following
-#' arguments: towstrat, type, towid, and weightid
-#' @param towstrat A vector of character values providing the column
-#' names for which you want compositions for.
-#' @param type A character value specifying which category to
-#' summarize by, i.e., `"length"` or `"Age"`.
-#' @param towid A vector of character values providing the column
-#' names that generate a unique id for each sample.
-#' @param weightid A character value giving the column name that
-#' holds the value to be summed for each type and strata.
-#' @param dropmissing A logical value supplied to the
-#' \code{drop} argument in \code{stats::aggregate}
-#' that specifies whether or not to keep all levels in the data
-#' even if there are no values to report for summaries. 
-#' @author Kelli F. Johnson
-#' @return A data frame in long form with a weight for each
-#' category included in the lengths or ages of interest by
-#' stratification. Stratifications are normally year and fishery
-#' and sex will always be included. If no sex is provided then it
-#' is assumed all are unsexed and will be returned as such.
+#' Create a long data frame to prepare compositions for [writeComps()]
+#' @param data A data frame. The data frame must have the column names that
+#'   are specified for the next four input arguments, `towstrat`, `type`, `towid`, and
+#'   `weightid`.
+#' @param towstrat A vector of character values providing the column names for
+#'   which you want compositions for.
+#' @param type A character value specifying which category to summarize by,
+#'   i.e., `"length"` or `"Age"`.
+#' @param towid A vector of character values providing the column names that
+#'   generate a unique id for each sample.
+#' @param weightid A character value giving the column name that holds the
+#'   value to be summed for each type and strata.
+#' @param dropmissing A logical value supplied to the `drop` argument in
+#'   [stats::aggregate()] that specifies whether or not to keep all levels in
+#'   the data even if there are no values to report for summaries.
+#' @export
+#' @describeIn getComps The actual workhorse behind [getComps()].
```

Comment:
`@describeIn` is new to me. From what I can tell, it tells roxygen that `getcomps_long()` is documented in the description of `getComps()`, so `getcomps_long.Rd` goes away and running 
`?PacFIN.Utilities::getcomps_long` sends you to the help page for `getComps()`. All that seems good.

Two trivial questions that it raises are 
1. why not just delete the roxygen comments at the top of `getcomps_long()`? Perhaps so that developers can see them and/or to facilitate moving away from `@describeIn` in the future?
2. Maybe it's better not to make `[getComps()]` a link since it the link only appears on the page which it links to:
![image](https://github.com/pfmc-assessments/PacFIN.Utilities/assets/4992918/03593a80-363c-4d02-baa0-0ac8f020f905)

Reply:
* `getComps()` is no longer a link, just code
* I used `describedIn` because I wanted the documentation for the input arguments of `getcomps_long()` to be visible to users when they looked up `getComps()` otherwise they may never know which arguments can be passed through from `getComps(...)`. Eventually, this will be better organized and there won't be a helper function with a hidden default argument 😮‍💨 but for right now I think combining the documentation is the 🏅 solution.